lib/index.js

```diff
@@ -469,3 +469,71 @@ const millisToMinutesAndSeconds = (millis) => {
     return `${minutes}:${(parseInt(seconds) < 10 ? "0" : "")}${seconds}.${milliseconds}`; //If seconds is less than 10 put a zero in front.
 };
 exports.millisToMinutesAndSeconds = millisToMinutesAndSeconds;
+/**
+ * Retrieves the day of the week from a given date string.
+ * @param {string} dateString - The input date string.
+ * @returns {string} - The day of the week as a string, or "Invalid date format" if the input is not a valid date.
+ */
+const getDayFromDate = (dateString) => {
+    const dateFormats = ["DD-MMM-YYYY", "DD/MM/YYYY", "DD/MMM/YYYY"];
+    let date = null;
+    for (let format of dateFormats) {
+        date = (0, exports.parseDate)(dateString, format);
+        if (date)
+            break;
+    }
+    if (!date) {
+        return "Invalid date format";
+    }
+    const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const dayOfWeek = date.getDay();
+    return days[dayOfWeek];
+};
+exports.getDayFromDate = getDayFromDate;
+/**
+ * Parses the input date string based on the specified format and returns a Date object.
+ * @param {string} dateString - The input date string.
+ * @param {string} format - The format string specifying the expected date format.
+ * @returns {Date} - The parsed Date object, or null if the input is not a valid date.
+ */
+const parseDate = (dateString, format) => {
```

Comment:
not a good function name, I can't tell what will this function do by just reading the name. @divyjainjd

Reply:
@jainarpit014 

- "parseDateByFormat"  will work ??
- as we are formatting the date string in this formats ("DD-MMM-YYYY", "DD/MM/YYYY", "DD/MMM/YYYY") and it will return a dateObject

Reply:
"strToDate" @divyjainjd